cmd/frontend/internal/app/ui/sveltekit/gen/gen_routes.go

```diff
@@ -0,0 +1,303 @@
+		if err != nil {
+			return err
+		}
+		if filepath.Base(path) == "+page.svelte" {
+			paths = append(paths, path)
+		}
+		return nil
+	})
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error walking directory: %v", err)
+		os.Exit(1)
+	}
+	return paths
+}
+
+var (
+	splitPattern = regexp.MustCompile(`\s+`)
+	tagPattern = regexp.MustCompile(`^\s*//\s+@sg\s+`)
+	groupPattern = regexp.MustCompile(`^\([^)]+\)$`)
+	paramPattern = regexp.MustCompile(`^(\[)?(\.\.\.)?(\w+)(?:=(\w+))?(\])?$`)
+	restParamPattern = regexp.MustCompile(`^\[\.\.\.(\w+)(?:=(\w+))?\]$`)
+	optionalParamPattern = regexp.MustCompile(`^\[\[(\w+)(?:=(\w+))?\]\]$`)
+)
+
+// extractTags finds lines of the forms
+//     // @sg tag1 tag2 tag3
+// and returns the tags found.
+func getRouteInfo(path string) (routeInfo, error) {
+	var info routeInfo
+
+	f, err := os.Open(path)
+	if err != nil {
+		return info, err
+	}
+	defer f.Close()
+
+	// All routes are opt-in by default
+	tagNames := []string{"EnableOptIn"}
+
+
+	scanner := bufio.NewScanner(f)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if tagPattern.MatchString(line) {
+			match := tagPattern.FindString(line)
+			tagNames = append(tagNames, splitPattern.Split(strings.TrimPrefix(line, match), -1)...)
+			break
+		}
+	}
+
+	for _, tag := range tagNames {
+		if !tags.IsTagValid(tag) {
+			return info, errors.Newf("Invalid tag '%s'. Valid tags: %s", tag, tags.AvailableTags())
+		}
+	}
+	info.Tags = tagNames
+
+	if err := scanner.Err(); err != nil {
+		return info, err
+	}
+
+	return info, nil
+}
+
+// Map SvelteKit specific parameter matchers to regular expressions. This is a "best effort" approach
+// because parameter matchers in SvelteKit are functions that can perform arbitrary logic.
+var paramMatchers = map[string]string{
+	"reporev": "/" + routevar.RepoPatternNonCapturing + `(?:@` + routevar.RevPatternNonCapturing + `)?`,
+}
+
+// This code follows the regex generation logic from
+// https://github.com/sveltejs/kit/blob/main/packages/kit/src/utils/routing.js
+func patternForRouteId(routeId string) (string, error) {
+	if routeId == "/" {
+		return "^/$", nil
+	}
+
+	b := strings.Builder{}
+	b.WriteByte('^')
+
+	for _, segment := range toSegments(routeId) {
+		if segment == "" {
+			continue
+		}
+
+		// [...rest]
+		if restParamPattern.MatchString(segment) {
+			matches := restParamPattern.FindStringSubmatch(segment)
+			if (len(matches) == 3) {
+				if matcher, ok := paramMatchers[matches[2]]; ok {
+					b.WriteString(matcher)
+					continue
+				}
+			}
+			b.WriteString(`(?:/.*)?`)
+			continue
+		}
+
+		// [[optional]]
+		if optionalParamPattern.MatchString(segment) {
+			b.WriteString(`(?:/[^/]+)?`)
+			continue
+		}
+
+		b.WriteByte('/')
+		// We don't use params within a segement, e.g.
+		// foo-[bar]-[[baz]], so for simplicity we don't support that.
+		if strings.Contains(segment, "[") {
+			return "", errors.New("params within a segment are not supported")
+		}
+
+		b.WriteString(regexp.QuoteMeta(segment))
```

Comment:
style nit: it's easier for me to understand if we write the leading `/` right before we write the name

```suggestion
		// We don't use params within a segement, e.g.
		// foo-[bar]-[[baz]], so for simplicity we don't support that.
		if strings.Contains(segment, "[") {
			return "", errors.New("params within a segment are not supported")
		}
		
		b.WriteByte('/')
		b.WriteString(regexp.QuoteMeta(segment))
```